OOP_Homework_2/RationalNumber.cs

```diff
@@ -0,0 +1,271 @@
+        {
+            return new RationalNumber(a) <= b;
+        }
+
+        // Возвращает наименьшее общее кратное
+        private static int getLeastCommonMultiple(int a, int b)
+        {
+            return a * b / getGreatestCommonDivisor(a, b);
+        }
+
+        /// <summary>
+        /// Метод создан для устранения повторяющегося кода в методах сложения и вычитания дробей.
+        /// Возвращает дробь, которая является результатом сложения или вычитания дробей a и b,
+        /// В зависимости от того, какая операция передана в параметр operation.
+        /// P.S. использовать только для сложения и вычитания
+        /// </summary>
+        /// <param name="a">дробь1</param>
+        /// <param name="b">дробь2</param>
+        /// <param name="operation"></param>
+        /// <returns></returns>
+        private static RationalNumber performOperation(RationalNumber a, RationalNumber b, Func<int, int, int> operation)
+        {
+            // Наименьшее общее кратное знаменателей
+            int leastCommonMultiple = getLeastCommonMultiple(a.denominator, b.denominator);
+            // Дополнительный множитель к первой дроби
+            int additionalMultiplierFirst = leastCommonMultiple / a.denominator;
+            // Дополнительный множитель ко второй дроби
+            int additionalMultiplierSecond = leastCommonMultiple / b.denominator;
+            // Результат операции
+            int operationResult = operation(a.numerator * additionalMultiplierFirst,
+            b.numerator * additionalMultiplierSecond);
+            return new RationalNumber(operationResult, a.denominator * additionalMultiplierFirst);
+        }
+
+        // Перегрузка оператора "+" для случая сложения двух дробей
+        public static RationalNumber operator +(RationalNumber a, RationalNumber b)
+        {
+            return performOperation(a, b, (int x, int y) => x + y);
+        }
+        // Перегрузка оператора "+" для случая сложения дроби с числом
+        public static RationalNumber operator +(RationalNumber a, int b)
+        {
+            return a + new RationalNumber(b);
+        }
+        // Перегрузка оператора "+" для случая сложения числа с дробью
+        public static RationalNumber operator +(int a, RationalNumber b)
+        {
+            return b + a;
+        }
+
+        // Перегрузка оператора "++"
+        public static RationalNumber operator ++(RationalNumber a)
+        {
+            return a + 1;
+        }
+
+        // Переопределение метода ToString
+        public override string ToString()
+        {
+            if (numerator == 0)
+            {
+                return "0";
+            }
+
+            if (numerator == denominator)
+            {
+                return "1";
+            }
+            if (denominator == 1)
+            {
+                return numerator.ToString();
+            }
+            return numerator + "/" + denominator;
```

Comment:
Помните, что конкатенация строк - это не самое лучшее решение. Лучше используйте строковую интерполяцию. А для большого числа объединяемых строк - `StringBuilder`
```suggestion
            return $"{numerator }/{denominator}";
```

Reply:
Строковая интерполяция имеет поддержку со стороны компилятора. Компилятор сможет выбрать наилучшую стратегию того, как реализовать то, что Вы хотите - формирование строки. Он может написать код с использованием `StringBuilder`, либо с помощью `string.Format("{0}/{1}, numerator, denomunator)`